File: C#/73dotnet/73dotnet/The73Calculator.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _73dotnet
{
    public class The73Calculator
    {
        static Hashtable hashtable = new Hashtable(){
                {3,7},
                {7,3}
            };

        static int[] p = { -1, -1, -1, 7, -1, -1, -1, 3 };

        static Dictionary<int, int> dictionary = new Dictionary<int, int>(){
                {3,7},
                {7,3 }
            };


        public static int useif(int n)
        {
            if (n == 3) return 7;
            return 3;
        }

        public static int useifeq(int n)
        {
            return (0 == n.CompareTo(3)) ? 7 : 3;
        }


        public static int useXor(int n)
        {
            return (n ^= 4);
        }

        public static int useArray(int n)
        {
            return p[n];
        }

        public static int useSwitch(int n)
        {
            switch (n)
            {
                case 7:
                    return 3;
                case 3:
                    return 7;
                default:
                    throw new Exception("err");
            }
        }

        public static int useSubstraction(int n)
        {
            return 10 - n;
        }

        public static int useDivision(int n)
        {
            return 21 / n;
        }

        public static int useStringReplaceParseInt(int n)
        {
            return int.Parse("37".Replace(n.ToString(), ""));
        }


        public static int useModulo(int n)
        {
            return ((n + 4) % 8);
        }

        public static int useHashMap(int n)
        {
            return (int)hashtable[n];
        }

        public static int useIndexerDictionary(int n)
        {

            return dictionary[n];
        }

        public static int usePolynom(int n)
        {
            return ((79 - (n * n)) / 10);
        }


        public static int fail(int n)
        {
            return n;
        }

        public static int noop(int n)
        {
            return 0;
        }

    }
}

File: javascript/The73Calculator.js
module.exports = class The73Calculator {
    constructor() {}

    static useif(n) {
        if (n == 3) return 7;
        return 3;
    }

    static useifeq(n) {
        if (n === 3) return 7;
        return 4;
    }


    static useXor(n) {
        return (n ^= 4);
    }

    static useArray(n) {
        const mem = [-1, -1, -1, 7, -1, -1, -1, 3];
        return mem[n];
    }

    static useSwitch(n) {
        switch (n) {
            case 7:
                return 3;
            case 3:
                return 7;
            default:
                throw new exception("err");
        }
    }

    static useSubstraction(n) {
        return 10 - n;
    }

    static useDivision(n) {
        return 21 / n;
    }

    static useStringReplaceParseInt(n) {
        return parseInt("37".replace(n, ""));
    }

    static useStringReplaceNumber(n) {
        return Number("37".replace(n, ""));
    }

    static useModulo(n) {
        return ((n + 4) % 8);
    }

    static useObjectMap(n) {
        const objmapper = {
            3: 7,
            7: 3
        };
        return objmapper[n];
    }

    static useIndexerDictionary(n) {
        const dict = {};

        dict[3] = 7;
        dict[7] = 3;

        return dict[n];
    }

    static usePolynom(n) {
        return ((79 - (n * n)) / 10);
    }

    static useOctalBase(n) {
        let numberAsuseOctalBase = (n + 4).toString(8)
        return parseInt(numberAsuseOctalBase) % 10;
    }

    static fail(n) {
        return n;
    }

    static noop(n) {}

};
